pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -256,18 +269,14 @@ func assignFlavors(log logr.Logger, requests []workload.PodSetResources, podSets
 		TotalBorrow: make(cache.FlavorResourceQuantities),
 		PodSets:     make([]PodSetAssignment, 0, len(requests)),
 		Usage:       make(cache.FlavorResourceQuantities),
-	}
-	if lastAssignment != nil {
-		assignment.LastState = *lastAssignment
-	} else {
-		assignment.LastState = workload.AssigmentClusterQueueState{
+		LastState: workload.AssigmentClusterQueueState{
```

Comment:
Why don't we set the `lastAssignment` to the `assignment.LastState` when the `lastAssignment` isn't nil?
IIUC, we can use the `lastAssignment` since the `lastAssignment` is the latest, not outdated here.